public/coffee/ide/editor/directives/aceEditor.coffee

```diff
@@ -374,6 +374,18 @@ define [
 					if scope.eventsBridge?
 						session.on "changeScrollTop", onScroll
 
+					hasLintingError = null
+					session.on('changeAnnotation', () ->
+						hasErrors = session
+							.getAnnotations()
+							.filter((annotation) -> annotation.type == 'error')
+							.length > 0
+
+						if (hasLintingError != hasErrors)
+							ide.$scope.$broadcast('ide:lintingError', hasErrors)
```

Comment:
We already have a reference to `$rootScope`, so we shouldn't need `ide` here.

Reply:
In fact, it should be possible to just trigger an event on our own `scope`, not even `$rootScope`

Reply:
See https://github.com/sharelatex/web-sharelatex/blob/2ef8d29a907259f5080473d93c881c80f0b5bec5/public/coffee/ide/editor/directives/aceEditor.coffee#L304 for an example, which also namespaces the event under the editor name, which is important because there are multiple Ace editors in the app (one is also used for the history view)

Reply:
Ah you're right. For some reason I thought pdf and editor were in different scopes. Also changed the event name to namespace under the editor.